woocommerce/woocommerce-ios
# Products: Added the variation number to the variations list, details page and updated progress view text when saving a variation

Closes #4846
Closes #4847

## Why
#4846 
 Variation number was not displayed in the Variation listing and details page.

#4847
Both Products and Variation details pages use the same controller `ProductFormViewController`,  So the message shown while saving variation changes was the same as that of product change. 

In this PR I added variation number in the variation listing and details page, also updated the progress view title and message when saving product variation.

## How
#4846
To add product variation number in variation listing page:
- Added `productVariation` of type `ProductVariation` in `ProductsTabProductViewModel` to get the variation ID (same as variation number)
- Above added `productVariation.productVariationID` is concatenated with cell name label to show variation number and name in variation listing
- 'nameLabel.numberOfLines' changed to 0 to avoid name clipping.
- Localization added for variation ID

To add product variation number in variation details page:
- Added `productionVariationID` in view model (`ProductFormViewModelProtocol`)
- Set navigation bar title in `updateNavigationBarTitle`
- If `viewModel.productionVariationID` is not `nil`, `updateNavigationBarTitle` function will set variation title.
- `viewModel.productionVariationID` becomes nil for production details page.

#4847
To update the texts in progress view page when saving a variation
- Added `saveVariation`, a new type in enum `SaveMessageType` to handle message type while showing progress view.
- Set messageType to `saveVariation` if the `viewModel` is `ProductVariationFormViewModel`.
- Based on the message type saving progress view will show a proper message for variation and product.

## Demo

#4648
https://user-images.githubusercontent.com/19687299/141671030-21d122a9-8d93-47a6-8eeb-cc2555e31e85.mov

#4847
https://user-images.githubusercontent.com/19687299/141671041-a8029f03-e14a-4aa5-852b-bc16b2f9d0fa.mov

## Testing Steps
#4846
- Open the Products tab
- Tap on any product with variations
- Tap on `Variations` option to show variation listing with variation number
- Tap on any variation item from variation listing to open the details page

#4847
- Open the Products tab
- Tap on any product with variations
- Tap on `Variations` option to open variation listing page
- Tap on any variation item from variation listing to open the details page
- Tap on `Description` or update any value
- Tap on `Save` button which appears on navigation bar

Update release notes:
